src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp

```diff
@@ -29,6 +31,30 @@
 
 using namespace mlir;
 
+//===----------------------------------------------------------------------===//
+// Insertion point for initialization instructions and the blocks used for
+// inserting the initialization and main code. These blocks will disappear
+// when the first canonicalization is performed because the init block
+// unconditionally branches into the second block. These blocks exist only for
+// the purpose of this optimization.
+// The support happens on a per function basis.
+//===----------------------------------------------------------------------===//
+
+typedef struct ONNXOperandsInitState {
+  Block *initBlock;
+  Block *mainBlock;
+  BranchOp branchInit;
+  llvm::SetVector<Value> *operandsInInitBlock;
+
+  ONNXOperandsInitState() {
+    operandsInInitBlock = new llvm::SetVector<Value>();
```

Comment:
I'd prefer making it a std::unique_ptr and manage deallocation automatically. The benefit is that it makes the ownership clear & enforced (since it can't be copied, we are also sure that there'll be no dangling pointer).

Reply:
Done